warehouse/ingest-core/src/main/java/datawave/ingest/util/WeightedValuePruningStrategy.java

```diff
@@ -134,7 +134,7 @@ private int calculateWeightings(final Integer[] originalStringLengths) {
                 int reducedScaleNgramCount = BloomFilterUtil.predictNGramCount((float) scalableOriginalStringLengths[i], this.maxNGramLength);
                 if (ratioOfTotalNgramCountsToMaxAllowed > 1.0f) {
                     reducedScaleNgramCount = Math.round(((float) reducedScaleNgramCount) * ratioOfTotalNgramCountsToMaxAllowed);
-                } else {
+                } else if (ratioOfTotalNgramCountsToMaxAllowed > 0.0f) {
```

Comment:
Perhaps this should be != 0.0f, and then we should consider what to do if it is 0.0f.

Reply:
The comments in the code seem to indicate that it's possible that `ratioOfTotalNgramCountsToMaxAllowed` could be a negative number (i.e. if `totalAllowedNgramsCount` were set to a negative number).  See the comment about this on line 46.

If negative numbers are allowed/expected, then this code change would affect how reducedScaleNgramCount is calculated.